simple_parsing/wrappers/field_wrapper.py

```diff
@@ -164,61 +169,27 @@ def get_arg_options(self) -> Dict[str, Any]:
             # automatically adds the (default: '123')
             _arg_options["help"] = " "
 
-        if utils.is_choice(self.field):
-            _arg_options["type"] = str
-            _arg_options["choices"] = list(self.choices)
+        if self.is_choice:
+            if isinstance(self.choices, dict):
+                _arg_options["type"] = str
+                _arg_options["choices"] = list(self.choices.values())
```

Comment:
Assuming from memory that `self.choices` was a dict before as well, here you're changing `arg_options["choices"]` from from being the keys of `choices` to instead be the values, right? Doesn't this cause any issues in tests?

Reply:
self.choices is now passed in from DataclassWrapper. Basically I think this code would be a lot cleaner and more flexible if it acted solely as a mechanism, leaving policy decisions to DCW. So at this point, FieldWrapper doesn't care if the choice froms from Literal, choice() or anywhere else. Maybe we can leverage the same code for enums too later on, but this is outside the scope of this PR.

Reply:
Sure, I agree, but that's not what I'm asking exactly: I'm asking, say the choices was a dict with a non-trivial type for the values, say something like:
```python
optimizer: Type[torch.optim.Optimizer] = choice({"adam": torch.optim.Adam, "sgd": torch.optim.SGD})
```
I'm asking if this change here will make the options to choose from, from the command-line, to be `"{<class Adam> <class SGD>}"` instead of just `{"adam", "sgd"}`

Reply:
Good catch, my bad, this does change the semantics. However, the dict field ends up getting overwritten by custom_args. Either way I decided to put all the code in FieldWrapper instead of splitting it. I'd still do it the way I suggested but that would require a refactor/rewrite and I don't want to end this PR with a mixed paradigm which is worse.